pkg/controllers/provisioningv2/rke2/plansecret/plansecret.go

```diff
@@ -58,7 +62,7 @@ func (h *handler) OnChange(key string, secret *corev1.Secret) (*corev1.Secret, e
 		}
 	}
 
-	if v, ok := node.PeriodicOutput["etcd-snapshot-list-s3"]; ok && v.ExitCode == 0 && len(v.Stdout) > 0 {
+	if v, ok := node.PeriodicOutput["etcd-snapshot-list-s3"]; ok && v.ExitCode == 0 && len(v.Stdout) > 0 && secret.Labels[rke2.InitNodeLabel] == "true" {
```

Comment:
Do we need this check if the command is only sent to the init node?

Reply:
Yes.

If an init node is re-elected, the old init node will still have the periodic output even if the instruction is not defined.